node/subsystem/src/jaeger.rs

```diff
@@ -0,0 +1,262 @@
+		if let Some(span) = src {
+			Self::Enabled(span)
+		} else {
+			Self::Disabled
+		}
+	}
+}
+
+impl From<mick_jaeger::Span> for JaegerSpan {
+	fn from(src: mick_jaeger::Span) -> Self {
+		Self::Enabled(src)
+	}
+}
+
+/// Shortcut for [`candidate_hash_span`] with the hash of the `Candidate` block.
+#[inline(always)]
+pub fn candidate_hash_span(candidate_hash: &CandidateHash, span_name: impl Into<String>) -> JaegerSpan {
+	hash_span(&candidate_hash.0, span_name)
+}
+
+/// Shortcut for [`hash_span`] with the hash of the `PoV`.
+#[inline(always)]
+pub fn pov_span(pov: &PoV, span_name: impl Into<String>) -> JaegerSpan {
+	hash_span(&pov.hash(), span_name)
+}
+
+/// Creates a `Span` referring to the given hash. All spans created with [`hash_span`] with the
+/// same hash (even from multiple different nodes) will be visible in the same view on Jaeger.
+#[inline(always)]
+pub fn hash_span(hash: &Hash, span_name: impl Into<String>) -> JaegerSpan {
+	INSTANCE.read_recursive().span(hash, span_name).into()
+}
+
+/// Stateful convenience wrapper around [`mick_jaeger`].
+pub enum Jaeger {
+	/// Launched and operational state.
+	Launched {
+		/// [`mick_jaeger`] provided API to record spans to.
+		traces_in: Arc<mick_jaeger::TracesIn>,
+	},
+	/// Preparation state with the necessary config to launch the collector.
+	Prep(JaegerConfig),
+	/// Uninitialized, suggests wrong API usage if encountered.
+	None,
+}
+
+impl Jaeger {
+	/// Spawn the jaeger instance.
+	pub fn new(cfg: JaegerConfig) -> Self {
+		Jaeger::Prep(cfg)
+	}
+
+	/// Spawn the background task in order to send the tracing information out via udp
+	#[cfg(target_os = "unknown")]
+	pub fn launch(self) -> result::Result<(), JaegerError> {
+		Ok(())
+	}
+
+	/// Spawn the background task in order to send the tracing information out via udp
+	#[cfg(not(target_os = "unknown"))]
+	pub fn launch(self) -> result::Result<(), JaegerError> {
+		let cfg = match self {
+			Self::Prep(cfg) => Ok(cfg),
+			Self::Launched{ .. } => {
+				return Err(JaegerError::AlreadyLaunched)
+			}
+			Self::None => Err(JaegerError::MissingConfiguration),
+		}?;
+
+		let jaeger_agent = cfg.agent_addr;
+
+		log::info!("🐹 Collecting jaeger spans for {:?}", &jaeger_agent);
+
+		let (traces_in, mut traces_out) = mick_jaeger::init(mick_jaeger::Config {
+			service_name: format!("{}-{}", cfg.node_name, cfg.node_name),
+		});
+
+
+		// Spawn a background task that pulls span information and sends them on the network.
+		let _handle = async_std::task::spawn::<_, result::Result<(), JaegerError>>(async move {
+			let udp_socket = async_std::net::UdpSocket::bind("127.0.0.1:0").await
+				.map_err(|e| JaegerError::PortAllocationError(e))?;
+
+			loop {
+				let buf = traces_out.next().await;
+				// UDP sending errors happen only either if the API is misused or in case of missing privilege.
+				if let Err(e) = udp_socket.send_to(&buf, jaeger_agent).await
+					.map_err(|e| JaegerError::SendError(e))
```

Comment:
Do we really need to map this error here? What is the benefit?

Reply:
A better log message, so you know what failed to send and where it comes from. Also, for consistency.